### ISSUE TITLE
Purge usage of malloc() and free()

### ISSUE THREAD

**Issue by tupaschoal** [OpenRCT2/OpenRCT2#11159]
OpenRCT2 historically used C and many places still use `malloc` and `free` for allocation and deallocation, which is far from ideal. The goal is to replace these with `new` and `delete`, but even better to wrap this content into one of `std::unique_ptr` or `std::shared_ptr`, using the corresponding `std::make_unique` and `std::make_shared` where applicable.

It's recommended to break into multiple PRs inside each of these folders in `src`, for ease of review and better progress:
- [ ] openrct2
- [ ] openrct2-cli
- [ ] openrct2-ui
- [ ] openrct2-win

**Comment by BasJansenDev** [comment]
Sounds like a good way for me to get familiar with the code. Mind if I attempt it?

**Comment by AaronVanGeffen** [comment]
This relates to our dependencies as well, most prominently the JSON library we use (cf. #9088). While the calls to malloc and free are wrapped by these dependencies, I'd rather avoid them entirely by using a C++ library.

**Comment by tupaschoal** [comment]
Sure, give it a spin @BasJansenDev

**Comment by IntelOrca** [comment]
@AaronVanGeffen Ideally we would use rapidjson as that is what the discord API uses but that does not use any exception handling. If you attempt to obtain a property that doesn't exist or a property of the wrong data type you will get a segfault. I'd much rather use a safer JSON library.

**Comment by AaronVanGeffen** [comment]
@IntelOrca I completely agree, fast as it may be, that's off the table IMO. Instead, I suggested [JSON for Modern C++](https://github.com/nlohmann/json#examples) in #9088, as it has lush syntax combined with good error handling.

**Comment by Vipulsh** [comment]
I would like to pick it up and give it a try. Will try to wrap it within smart pointers as that will be a good learning and understanding for me too

**Comment by tupaschoal** [comment]
Ok, take it slow and try to start from a file you can compile and that doesn't overlap with the PR above

**Comment by rzhao271** [comment]
Just realized I'm supposed to comment on an issue to claim it, though the commits above look abandoned.
I'd like to work on this issue. 
I submitted a PR above to remove some occurrences of `free()` in the `openrct2-ui` folder, but most of the `malloc()` and `free()` calls are in the `openrct2` folder.
